src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
Thanks for the link, but I'm still not following. Does `revealed` mean that is the one currently active in the editor? If yes, that would be state inside the controller, right? You said the controller needs to clear it, but if it doesn't have to be cleared from the outside (online from inside the controller), then I don't see why we need to pass a wrapper object to the controller